widget/embedded/src/components/Quote/QuoteCostDetails.tsx

```diff
@@ -0,0 +1,236 @@
+        time={totalTime}
+        steps={steps}
+        tooltipGas={i18n.t('View more info')}
+      />
+
+      <Modal
+        container={document.getElementById(RANGO_SWAP_BOX_ID) || document.body}
+        open={open}
+        header={
+          <ModalHeader>
+            <Typography variant="title" size="medium">
+              {i18n.t('Gas & Fee Explanation')}
+            </Typography>
+
+            <IconButton onClick={() => setOpen(false)} variant="ghost">
+              <CloseIcon color="gray" size={14} />
+            </IconButton>
+          </ModalHeader>
+        }
+        onClose={() => setOpen(false)}>
+        <ModalContainer>
+          <Typography variant="title" size="small">
+            {i18n.t('Details')}
+          </Typography>
+          <Divider size={12} />
+          {swaps.map((swap) =>
+            swap.fee.map((fee, i) => {
+              const key = `fee-${i}`;
+              const usdValue = getUsdSwapsFeeOfStep(swap, 'Network Fee');
+              return (
+                fee.name === 'Network Fee' && (
+                  <>
+                    <FeeSection key={key}>
+                      <Typography
+                        variant="label"
+                        size="medium"
+                        color="neutral600">
+                        {i18n.t('Network Fee')}
+                      </Typography>
+                      <Tooltip
+                        content={formatTooltipNumbers(fee.amount)}
+                        container={tooltipContainer}>
+                        <Typography variant="label" size="medium">
+                          {numberToString(
+                            fee.amount,
+                            GAS_FEE_MIN_DECIMALS,
+                            GAS_FEE_MAX_DECIMALS
+                          )}{' '}
+                          {fee.asset.symbol} ($
+                          {numberToString(
+                            usdValue,
+                            USD_VALUE_MIN_DECIMALS,
+                            USD_VALUE_MAX_DECIMALS
+                          )}
+                          )
+                        </Typography>
+                      </Tooltip>
+                    </FeeSection>
+                    <Divider size={8} />
+                  </>
+                )
+              );
+            })
+          )}
+          <Divider size={12} />
+          <FeeSection>
+            <Typography variant="label" size="medium">
+              {i18n.t('Total Payable Fee')}
+            </Typography>
+            <Typography variant="label" size="medium">
+              $
+              {numberToString(
+                totalNetworkFeeInUsd,
+                USD_VALUE_MIN_DECIMALS,
+                USD_VALUE_MAX_DECIMALS
+              )}
+            </Typography>
+          </FeeSection>
+          <Divider size={12} />
+          <Line />
+          <Divider size={12} />
+          <Collapsible.Root
```

Comment:
In the ` widget/embedded/src/components/ConfirmWalletsModal/ConfirmWallets.styles.ts` we have a collapsible component with transition to increase/decrease height. I think we can define it as a separated component to re-use it here to have a transition for showing payable/non-payable collapse.

Reply:
Because of the differences in their structures, I think it cannot be implemented.

Reply:
Ok. Please just add a transition on open/close collapsible component.